cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -530,9 +533,9 @@ gdv_boolean compare_lower_strings(const char* base_str, gdv_int32 base_str_len,
 // Try to cast the received string ('0', '1', 'true', 'false'), ignoring leading
 // and trailing spaces, also ignoring lower and upper case.
 FORCE_INLINE
-gdv_boolean castBIT_utf8(gdv_int64 context, const char* data, gdv_int32 data_len) {
+gdv_boolean castBIT_utf8(void* context_ptr, const char* data, gdv_int32 data_len) {
   if (data_len <= 0) {
-    gdv_fn_context_set_error_msg(context, "Invalid value for boolean.");
+    gdv_fn_context_set_error_msg(context_ptr, "Invalid value for boolean.");
```

Comment:
Not related to this PR. But it is a generic comment on not enough information in error messages through out Gandiva, which make it very difficult to understand why Gandiva rejected a request.

In this case, the error message should have more details: e.g 'Invalid value for boolean; bad length \<length\>'